index.js

```diff
@@ -308,15 +308,50 @@ SendStream.prototype.hasTrailingSlash = function hasTrailingSlash () {
 }
 
 /**
+ * Returns `true` if any request preconditions are failing
+ *
+ * @return {Boolean}
+ * @api private
+ */
+
+SendStream.prototype.isFailingPrecondition = function isFailingPrecondition (stat) {
+  // if-match
+  var match = this.req.headers['if-match']
+  var etag = this.res._headers['etag']
+  var strongEtag = /^(?!W\/)/
+
+  if (match) {
+    if (strongEtag.test(etag)) {
+      return !match.split(/ *, */).some(function (match) {
+        return match === '*' || match === etag
+      })
+    }
```

Comment:
It also seems to be missing on the way `*` works when this module is about to send a 404. It seems to me that if the request says `If-Match: "foo"`, then this module MUST 412 on the response if it would have normally 404ed from:

> If the field-value is "*", the condition is false if the origin server does not have a current representation for the target resource.